src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -110,7 +112,8 @@
 
     private List<PodImagePullSecret> imagePullSecrets = new ArrayList<PodImagePullSecret>();
 
-    private transient List<ToolLocationNodeProperty> nodeProperties;
+    @SuppressFBWarnings(value = "SE_BAD_FIELD")
+    private List<ToolLocationNodeProperty> nodeProperties = new ArrayList<>();
```

Comment:
on upgrade this will be null because the xml does not have the option `nodeProperties` entry, it should be checked in `getNodeProperties`